src/Sarif.Viewer.VisualStudio/Tags/ISarifLocationTaggerService.cs

```diff
@@ -1,22 +1,28 @@
-﻿// Copyright (c) Microsoft. All rights reserved. 
-// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.Sarif.Viewer.Tags
 {
-    using System;
-    using System.Runtime.InteropServices;
-
     [Guid("774C62D3-203A-493C-8801-BEA59FE46CA1")]
     internal interface ISarifLocationTaggerService
     {
         /// <summary>
-        /// Causes a tags changed notification to be sent out from all known taggers.
+        /// Causes a tags changed notification to be sent out from all known taggers associated
+        /// with the specified <see cref="ITextBuffer"/>.
         /// </summary>
+        /// <param name="textBuffer">
+        /// The text buffer of interest, or null if the notification should be set to all taggers.
+        /// </param>
         /// <remarks>
-        /// As an example, this method is called to send a tags changed notification when a "text view" is already open and visible
-        /// and a tagger is active for that "text view" when a SARIF log is loaded or cleared.
+        /// As an example, this method is called to send a tags changed notification when a "text
+        /// view" is already open and visible and a tagger is active for that "text view" when a
+        /// SARIF log is loaded or cleared.
         /// </remarks>
-        void RefreshAllTags();
+        void RefreshAllTags(ITextBuffer textBuffer = null);
```

Comment:
> = null [](start = 50, length = 7)

My personal preference is no default parameters because it's too easy to make a mistake... That's just me though, not sure what the coding convention is.

Reply:
Per discussion, merging without change.

---
In reply to: [516216441](https://github.com/microsoft/sarif-visualstudio-extension/pull/245#discussion_r516216441) [](ancestors = 516216441)